src/helpers/scaffoldProject.ts

```diff
@@ -26,30 +27,62 @@ export const scaffoldProject = async ({
   const spinner = ora(`Scaffolding in: ${projectDir}...\n`).start();
 
   if (fs.existsSync(projectDir)) {
-    if (fs.readdirSync(projectDir).length === 0) {
+    const userFilesSet = new Set(getAllFiles(projectDir));
+    const srcFiles = [...getAllFiles(srcDir), ".gitignore"]; // template files
+    const conflictingFiles = srcFiles.filter((element) =>
+      userFilesSet.has(element),
+    );
+
+    if (userFilesSet.size === 0) {
       spinner.info(
         `${chalk.cyan.bold(projectName)} exists but is empty, continuing...\n`,
       );
+    } else if (conflictingFiles.length === 0) {
+      spinner.info(
+        `${chalk.cyan.bold(
+          projectName,
+        )} is not empty, but no conflicting files found, continuing...\n`,
+      );
     } else {
+      let conflictingFilesStr = "";
+      for (const file of conflictingFiles) {
+        conflictingFilesStr += `\n  ${chalk.cyan.bold(
+          projectName,
+        )}${chalk.reset.red(path.sep + file)}`;
+      }
+
       spinner.stopAndPersist();
-      const { overwriteDir } = await inquirer.prompt<{ overwriteDir: boolean }>(
-        {
-          name: "overwriteDir",
-          type: "confirm",
-          message: `${chalk.redBright.bold("Warning:")} ${chalk.cyan.bold(
-            projectName,
-          )} already exists and isn't empty. Do you want to overwrite it?`,
-          default: false,
-        },
-      );
-      if (!overwriteDir) {
+      const { overwriteDir } = await inquirer.prompt<{ overwriteDir: string }>({
```

Comment:
Making this string type safe will help preventing typos if modified later.

```suggestion
      const { overwriteDir } = await inquirer.prompt<{ overwriteDir: "clear" | "abort" | "overwrite" }>({
```